src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseAsyncMethodInAsyncContextTests.cs

```diff
@@ -1243,6 +1244,64 @@ End Module
             await CreateVBTestAndRunAsync(testVB);
         }
 
+        [Fact]
+        [WorkItem(6684, "https://github.com/dotnet/roslyn-analyzers/issues/6684")]
+        public Task DbContextAdd_NoDiagnostic()
+        {
+            return new VerifyCS.Test
+            {
+                TestCode = @"
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+
+class Test {
+    public async Task RunAsync(DbContext ctx) {
+        ctx.Add(1);
+    }
+}",
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net70.WithPackages(ImmutableArray.Create(new PackageIdentity("Microsoft.EntityFrameworkCore", "7.0.8")))
+            }.RunAsync();
+        }
+
+        [Fact]
+        [WorkItem(6684, "https://github.com/dotnet/roslyn-analyzers/issues/6684")]
+        public Task DbContextAddRange_NoDiagnostic()
+        {
+            return new VerifyCS.Test
+            {
+                TestCode = @"
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+
+class Test {
+    public async Task RunAsync(DbContext ctx) {
+        ctx.AddRange(1, 2);
+    }
+}",
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net70.WithPackages(ImmutableArray.Create(new PackageIdentity("Microsoft.EntityFrameworkCore", "7.0.8")))
+            }.RunAsync();
+        }
+
+        [Fact]
+        [WorkItem(6684, "https://github.com/dotnet/roslyn-analyzers/issues/6684")]
+        public Task DbContextFactoryCreateDbContext_Diagnostic()
+        {
+            return new VerifyCS.Test
+            {
+                TestCode = @"
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+
+class Test {
+    public async Task RunAsync(IDbContextFactory<DbContext> factory) {
+        var context = {|#0:factory.CreateDbContext()|};
+    }
+}",
+                ExpectedDiagnostics = { new DiagnosticResult(UseAsyncMethodInAsyncContext.Descriptor).WithLocation(0).WithArguments("IDbContextFactory<DbContext>.CreateDbContext()", "IDbContextFactory<DbContext>.CreateDbContextAsync(CancellationToken)") },
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net70.WithPackages(ImmutableArray.Create(new PackageIdentity("Microsoft.EntityFrameworkCore", "7.0.8")))
```

Comment:
Lets use static readonly field for `ImmutableArray.Create(new PackageIdentity("Microsoft.EntityFrameworkCore", "7.0.8")` and reuse for each test